src/navigator/NavigationService.ts

```diff
@@ -45,6 +47,23 @@ async function ensureNavigator() {
   }
 }
 
+export const popToScreen: SafeNavigate = (...args) => {
+  const [routeName] = args
+  ensureNavigator()
+    .then(() => {
+      Logger.debug(`${TAG}@popToScreen`, `Dispatch ${routeName}`)
+      while (
+        navigationRef.current?.canGoBack() &&
+        navigationRef.current?.getCurrentRoute()?.name !== routeName
+      ) {
+        navigationRef.current?.dispatch(StackActions.pop())
+      }
```

Comment:
Feels a bit hacky 😅 
No other way to make this work?

Reply:
I'll take a  look at using a reset and setting the stack that way, but with my initial attempts it wasn't working.